### PR TITLE
StartupServices Support - Service Parameter Config Template addition

### PR DESCRIPTION
**Pull request by sunil-indoria**

### CODE REVIEW

File: properties/service_fabric_common.props [hunk 1]
```diff
@@ -31,7 +31,7 @@
     <!-- TODO: Versions numbers are changed here manually for now, Integrate this with GitVersion. -->
     <MajorVersion>5</MajorVersion>
     <MinorVersion>0</MinorVersion>
-    <BuildVersion>5</BuildVersion>
```
Comment (PavanKunapareddyMSFT) [inline properties/service_fabric_common.props:34]: Please avoid making build version changes in the same PR as other changes. Having only your workitem changes in the PR would work better for cherrypicking the commit to the release branch.

Comment (PavanKunapareddyMSFT) [reply properties/service_fabric_common.props:34]: Please revert the change in this file

**Review by PavanKunapareddyMSFT — CHANGES_REQUESTED** [review]

File: src/ModelXSD/StartupServicesModel.xsd [hunk 2]
```diff
@@ -6,6 +6,11 @@
     <xs:documentation>Microsoft Azure Service Fabric StartupServices Model. Version: V1_RTM_3 [MajorVersion_Milestone_ChangeNo]</xs:documentation>
   </xs:annotation>
   <xs:element name="StartupServicesManifest" type="StartupServicesManifestType"/>
+  <xs:element name="StartupServices" type="StartupServiceInstanceDefinitionType">
+    <xs:annotation>
+      <xs:documentation>Service Instance specific information like service parameter values used to create service. Parameter values in this file overrides the default parameter values defined in startup service Manifest.</xs:documentation>
```
Comment (anantshankar17) [inline src/ModelXSD/StartupServicesModel.xsd:11]: List of service instance.

File: src/ModelXSD/StartupServicesModel.xsd [hunk 3]
```diff
@@ -649,4 +654,44 @@
...
+    <xs:sequence>
+      <xs:element name="Parameters">
+        <xs:annotation>
+          <xs:documentation>List of service parameters for the application as defined in startup service manifest and their respective values.</xs:documentation>
+        </xs:annotation>
+        <xs:complexType>
+          <xs:sequence>
+            <xs:element name="Parameter" minOccurs="0" maxOccurs="unbounded">
+              <xs:complexType>
+                <xs:attributeGroup ref="StartupNameValuePair"/>
```
Comment (anantshankar17) [inline src/ModelXSD/StartupServicesModel.xsd:689]: what is startup name ?

File: src/ModelXSD/StartupServicesModel.xsd [hunk 4]
```diff
@@ -6,6 +6,11 @@
     <xs:documentation>Microsoft Azure Service Fabric StartupServices Model. Version: V1_RTM_3 [MajorVersion_Milestone_ChangeNo]</xs:documentation>
   </xs:annotation>
   <xs:element name="StartupServicesManifest" type="StartupServicesManifestType"/>
+  <xs:element name="StartupServices" type="StartupServiceInstanceDefinitionType">
```
Comment (anantshankar17) [inline src/ModelXSD/StartupServicesModel.xsd:9]: To prevent further iterations, can you confirm if this is "StartupServices" or just "Services" ?

**Review by anantshankar17 — CHANGES_REQUESTED** [review]

File: src/ModelXSD/StartupServicesModel.xsd (hunk 4 above)
Comment (sunil-indoria) [reply src/ModelXSD/StartupServicesModel.xsd:9]: it is StartupServices

File: src/ModelXSD/StartupServicesModel.xsd (hunk 3 above)
Comment (sunil-indoria) [reply src/ModelXSD/StartupServicesModel.xsd:689]: Parameters inside ServiceConfig file are of type Name and Value pair. This type is been defined as StartupNameValuePair  to separate it out from the NameValuePair defined in ServiceFabricServiceModel.xsd

**Review by PavanKunapareddyMSFT — APPROVED** [review]
LGTM

File: src/ModelXSD/StartupServicesModel.xsd (hunk 3 above)
Comment (anantshankar17) [reply src/ModelXSD/StartupServicesModel.xsd:689]: should it be StartupServiceNameValuePair

**Review by anantshankar17 — APPROVED** [review]
fix the minor comments..

**Review by anantshankar17 — APPROVED** [review]

File: src/ModelXSD/StartupServicesModel.xsd (hunk 3 above)
Comment (sunil-indoria) [reply src/ModelXSD/StartupServicesModel.xsd:689]: ServicesManifest  has been termed as StartupServicesManifest, similarly Startup has been added prefix to distinguish in other types as well in XSD e.g. StartupStatelessServiceType

**Review by mevora-msft — APPROVED** [review]

**Pull request merged**